code/datums/action.dm

```diff
@@ -204,6 +204,61 @@
 /datum/action/item_action/toggle_light
 	name = "Toggle Light"
 
+/datum/action/item_action/toggle_laser_sight
+	name = "Toggle Laser Sight"
+	icon_icon = 'icons/obj/guns/attachment.dmi'
+	button_icon_state = "laser_sight"
+	var/obj/item/attachment/laser_sight/att
+
+/datum/action/item_action/toggle_laser_sight/Trigger()
+	if(!att)
+		if(istype(target, /obj/item/gun))
+			var/obj/item/gun/H = target
+			for(var/obj/item/attachment/A in H.current_attachments)
+				if(istype(A, /obj/item/attachment/laser_sight))
+					att = A
+					break
+	if(att)
+		att.toggle_on()
```

Comment:
```suggestion
	att?.toggle_on()
```
this is what conditionals are for, not using many useless statements that make things harder to read

Reply:
don't have to be rude about it

Reply:
just explaining it, maybe came out the wrong way